### PR TITLE
Fix request header for Rack 1.x

### PR DESCRIPTION
**Pull request by lloeki**
Rack 1.x lacks a each_header method, adjust for it.

### CODE REVIEW

File: lib/datadog/appsec/contrib/rack/request.rb [hunk 1]
```diff
@@ -9,8 +9,14 @@ def self.query(request)
...
 
           def self.headers(request)
-            request.each_header.each_with_object({}) do |(k, v), h|
-              h[k.gsub(/^HTTP_/, '').downcase.tr('_', '-')] = v if k =~ /^HTTP_/
+            if request.respond_to?(:each_header)
+              request.each_header.each_with_object({}) do |(k, v), h|
+                h[k.gsub(/^HTTP_/, '').downcase.tr('_', '-')] = v if k =~ /^HTTP_/
+              end
+            else
+              request.env.select { |k, _v| k =~ /^HTTP_/ }.each_with_object({}) do |(k, v), h|
```
Comment (ivoanjo) [inline lib/datadog/appsec/contrib/rack/request.rb:17]: Minor: This `select` doesn't seem very useful, given we repeat the check anyway in the proc.

**Review by ivoanjo — COMMENTED** [review]
I know we don't have specs for a lot of things in appsec, but can we start by adding some here?

As an alternative, we should at leave a comment or an indication in the code to explain why `:each_header` may not be around.

File: lib/datadog/appsec/contrib/rack/request.rb (hunk 1 above)
Comment (lloeki) [reply lib/datadog/appsec/contrib/rack/request.rb]: You, sir, are absolutely correct. I don't know what I've been thinking, adding a useless array allocation just like that!

**Comment by lloeki** [comment]
> I know we don't have specs for a lot of things in appsec, but can we start by adding some here?

Comment added, but I'll try to add some.

My current priority is [having the system tests running and passing](https://github.com/DataDog/system-tests/pull/163) for [all suppported versions of rubies and frameworks](https://github.com/DataDog/system-tests-apps-ruby), adding proper specs for AppSec is my next task in line.

**Review by ivoanjo — APPROVED** [review]
👍  I can live with only having test coverage for this via the system tests for now.

(P.s.: Something tells me CI is broken due to a new rspec version)

**Review by marcotc — APPROVED** [review]
Looks good, just a minor comment. Feel free to merge.

**Comment by lloeki** [comment]
Alright, I got ahead of myself and adding the specs is turning into a [full-blown PR of its own](https://github.com/DataDog/dd-trace-rb/pull/1899).

Allow me to not digress too much and merge this so that I can unblock dependent PRs in the other repos.

**Pull request merged**